tests/unittests/test_init.py

```diff
@@ -0,0 +1,62 @@
+from singer.catalog import Catalog
+import unittest
+from unittest import mock
+from tap_quickbooks import main, do_discover
+from parameterized import parameterized
+
+class Parse_Args:
+    """Mocked parse args"""
+
+    def __init__(self, discover=False, state=None, catalog=None, properties=None, config_path=None) -> None:
+        self.discover = discover
+        self.state = state
+        self.config = {}
+        self.catalog = catalog
+        self.properties = properties
+        self.config_path = config_path
+
+class TestQuickbooksInit(unittest.TestCase):
+
+    @parameterized.expand([  # test_name, [discover_flag, catalog_flag], [do_discover, do_sync]
+        ['do_discover_called', [True, True], [True, False]],
+        ['do_sync_called', [False, True], [False, True]],
+        ['do_discover_and_do_sync_called', [False, False], [True, True]]
+    ])
+    @mock.patch('singer.parse_args')
+    @mock.patch('tap_quickbooks.write_catalog')
+    @mock.patch('tap_quickbooks.do_discover')
+    @mock.patch('tap_quickbooks.do_sync')
+    @mock.patch('tap_quickbooks.QuickbooksClient')
+    def test_init(self, test_name, test_data, exp, mocked_client, mocked_sync, mocked_discover, mock_catalog, mock_args):
```

Comment:
Is this the correct order for the decorators? If so, is there a way to make this more clear? test_init has arguments in an order that doesn't quite match the above given that the parameterized expand would be test_name, test_data, exp but then client through mocked_args decorators are in reversed order from the function args